lib/spack/docs/packages_yaml.rst

```diff
@@ -163,6 +161,55 @@ In addition, paths from ``extra_rpaths`` are added as library search paths for t
 In the example above, both ``/usr/lib/gcc`` and ``/usr/lib/unusual_gcc_path`` would be added as rpaths to the linker, and ``-L/usr/lib/unusual_gcc_path`` would be added as well.
 
 
+Specifying dependencies among external packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Dependency relationships among external packages can also be be modeled in your ``packages.yaml`` configuration file.
+For instance, the following configuration describes an ``mpileaks`` external that depends on ``callpath`` and on ``mpich``:
+
+.. code-block:: yaml
+
+   # Specification for the following DAG:
+   #
+   # o mpileaks@2.3
+   # |\
+   # | o callpath@1.0
+   # |/
+   # o mpich@3.0.4
+   packages:
+     mpileaks:
+       externals:
+       - spec: "mpileaks@2.3~debug+opt"
+         prefix: /user/path
+         dependencies:
+         - external_id: callpath_id
```

Comment:
if `deptypes` is required, then maybe

```
<id>:
  deptypes: link
```

to save a few characters and enforce the constraint of one edge between two nodes?